mysql-test/suite/innodb/t/xtradb_compressed_columns_with_dictionaries.test

```diff
@@ -66,26 +66,24 @@ SET @long_dictionary_name = REPEAT('d', 64);
 
 # max dictionary data size is 32506 bytes - check if an error
 # is reported if this number is exceeded
+# From 8.0, it is BLOB and size is unlimited. No error for long_dictionary_data
```

Comment:
I see. Let me debug and find out why it is not giving error in 8.0

Reply:
atleast while creating compression dictionary, it is just about storing data into table. The size can be unlimited
"
The implementation may silently use a subset of the provided dictionary if the dictionary cannot fit in the current window associated with stream (see deflateInit2_()). 
"
So the window size for zlib is maxed at 32K ?

Reply:
```
int deflateSetDictionary(z_streamp stream, const Bytef * dictionary, uInt dictlen);

Description

The deflateSetDictionary() function shall initialize the compression dictionary associated with stream using the dictlen bytes referenced by dictionary.

The implementation may silently use a subset of the provided dictionary if the dictionary cannot fit in the current window associated with stream (see deflateInit2_()). The application should ensure that the dictionary is sorted such that the most commonly used strings occur at the end of the dictionary.

``

Reply:
@satya-bodapati The main concern here is the user experience.
Even though, according to the documentation, it is OK to call `deflateSetDictionary()` with large (longer than windowSize) data passed as a dictionary, just silently ignoring the part of  this dictionary may not be very user-friendly and may rise a lot of questions. What is even more confusing is that in case of exceeding this 32506 limit, not the bytes at the end but the ones at the beginning will be ignored.

So, I suggest, as of right now we only have single compression algorithm (zlib) available for compression columns, we leave this 32506 compression dictionary size limit as is. In other words, calling `CREATE COMPRESSION_DICTIONARY()` with data longer than 32506 bytes should return en error.

However, I admit that in future we may have additional compression algorithms (like lz4, zstd, etc.) for which this limitation can be different. So, I suggest, we review/remove this limitation then.

Reply:
Ok @percona-ysorokin I will limit the dictionary size